apis/python/src/tiledbsoma/tiledb_array.py

```diff
@@ -177,3 +177,27 @@ def _chunk_is_contained_in(
                 return False
 
         return True
+
+    def _consolidate_and_vacuum_fragment_metadata(self) -> None:
+        """
+        This post-ingestion helper consolidates and vacuums fragment metadata -- this is quick to
+        do, and positively impacts query performance.  It does _not_ consolidate bulk array data,
+        which is more time-consuming and should be done at the user's opt-in discretion.
+        """
+
+        if self.uri.startswith("tiledb://"):
+            # At present we need to look up the S3 path. This can only be done via
+            # tiledb.cloud.info(self.uri). Yet this open-source package does not take a dependency
+            # on the tiledb.cloud package.
+            return
+
+        for mode in ["fragment_meta", "commits", "array_meta"]:
```

Comment:
I don't think we need to consolidate the `array_meta`. That should only written once per write so there isn't a need to consolidate. Multiple array metadata files will exist only across multiple independent writes so we can keep them for time traveling.